randomization/distribution_sample_generator.h

```diff
@@ -0,0 +1,53 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#ifndef DISTBENCH_RANDOMIZATION_RANDOM_SAMPLE_GENERATOR_H_
+#define DISTBENCH_RANDOMIZATION_RANDOM_SAMPLE_GENERATOR_H_
+
+#include <array>
+#include <chrono>
+#include <functional>
+#include <iostream>
+#include <random>
+
+#include "absl/status/statusor.h"
+#include "glog/logging.h"
+#include "randomization/distribution_config.pb.h"
+
+namespace distbench {
+
+absl::Status ValidateDistributionConfig(DistributionConfig& config);
+
+class DistributionSampleGenerator {
+ public:
+  ~DistributionSampleGenerator(){};
+
+  void Initialize(const DistributionConfig& config);
```

Comment:
Why doesn't this return an error code if given a bad input?

Reply:
Changed the type to absl::Status.